### PR TITLE
docs(ReadMe): Updated readMe & indexed fields on events.

### PR DESCRIPTION
**Pull request by Abhishekkochar**
#### What this PR does / why we need it:
As per the title, Added sections for sending and removing funds on bridge contract. 
There is also an guide on how to add owners to gnosis safe .

Also, indexed the events fields. Upto 3 fields can be indexed per event. 
Indexed fields are easy to find by off chain tools. 

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
 ### BridgeProxy
 
 BridgeProxy Contract addrress: [0x93fE70235854e7c97A5db5ddfC6eAAb078e99d3C](https://goerli.etherscan.io/address/0x93fE70235854e7c97A5db5ddfC6eAAb078e99d3C)
+
+## Fund Bridge with ETHER and ERC20 tokens
+
+### Add funds
```
Comment (Abhishekkochar) [inline README.md:133]: @cuongquangnam , can you please quickly double check this?

File: README.md [hunk 2]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
+
+Anyone can send fund to the bridge contract, ideally in this case, this should be done by `Cake`. If tokens send by other addresses, those tokens will be unaccounted for. In this case, Admin will have to manually refund the
+
+Admins can send the ETHER and ERC20 tokens via the `transfer(address _to, uint256 _amount)` function.
+
+Funding the bridge contract would ideally be done on lunch (Not decided on specific numbers)
```
Comment (cuongquangnam) [inline README.md:139]: on launch right, at first sight, I thought we gonna do funding while eating

File: README.md [hunk 3]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
 
 BridgeProxy Contract addrress: [0x93fE70235854e7c97A5db5ddfC6eAAb078e99d3C](https://goerli.etherscan.io/address/0x93fE70235854e7c97A5db5ddfC6eAAb078e99d3C)
+
+## Fund Bridge with ETHER and ERC20 tokens
+
+### Add funds
+
+Anyone can send fund to the bridge contract, ideally in this case, this should be done by `Cake`. If tokens send by other addresses, those tokens will be unaccounted for. In this case, Admin will have to manually refund the
+
+Admins can send the ETHER and ERC20 tokens via the `transfer(address _to, uint256 _amount)` function.
```
Comment (cuongquangnam) [inline README.md:137]: for the case of ether, it will be a simple transfer in metamask

File: README.md (hunk 2 above)
Comment (Abhishekkochar) [reply README.md:139]: oh that's one of a kind typo hahaha

File: README.md (hunk 3 above)
Comment (Abhishekkochar) [reply README.md:137]: Yes, however the underlying function will be `transfer` regardless of ETHER and ERC20 via metamask

File: README.md [hunk 4]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
 ### BridgeProxy
 
 BridgeProxy Contract addrress: [0x93fE70235854e7c97A5db5ddfC6eAAb078e99d3C](https://goerli.etherscan.io/address/0x93fE70235854e7c97A5db5ddfC6eAAb078e99d3C)
+
+## Fund Bridge with ETHER and ERC20 tokens
```
Comment (weiyuan95) [inline README.md:131]: There's no more ETH, only wETH 😛

File: README.md [hunk 5]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
+
+### Add funds
+
+Anyone can send fund to the bridge contract, ideally in this case, this should be done by liquidity providers. If tokens send by other addresses straight to the contract, those tokens will be unaccounted for. In this case, Admin will have to manually process the refund.
```
Comment (weiyuan95) [inline README.md:135]:
```suggestion
Anyone can send funds to the bridge contract. Ideally, this should be done by liquidity providers. If there are tokens sent by other addresses to the contract, those tokens will be unaccounted for.
```

Removing the part about processing refunds. Such cases will be the exception - not the norm.

File: README.md [hunk 6]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
+
+Admins can send the ETHER and ERC20 tokens via the `transfer(address _to, uint256 _amount)` function or utilizing wallet such as Metamask.
```
Comment (weiyuan95) [inline README.md:137]:
```suggestion
Admins can send the ERC20 tokens via the `transfer(address _to, uint256 _amount)` function or utilizing wallets such as Metamask.
```

File: README.md [hunk 7]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
+
+Funding the bridge contract would ideally be done upon launch (Not decided on specific numbers)
```
Comment (weiyuan95) [inline README.md:139]:
```suggestion
```

No need to mention this!

File: README.md [hunk 8]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
+
+### Withdraw funds
+
+ETHER and ERC20 can be withdrawn by the Admin address only via `withdrawEth(uint256 amount)` and `withdraw(address _tokenAddress, uint256 amount)` functions respectively.
```
Comment (weiyuan95) [inline README.md:143]:
```suggestion
### Withdrawing funds

ERC20 tokens can be withdrawn by the Admin address only via the `withdraw(address _tokenAddress, uint256 amount)` function.
```

Comment (weiyuan95) [reply README.md:143]: This will need to be updated when #277 implementation is added.

File: README.md [hunk 9]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
+
+## Admin and Operational addresses - Gnosis safe
+
+Admin and Operational addresses will be Gnosis safes, ideally will be with at least 3 owners. In case of 3 addresses, 2 approval ( this can be set when making a safe or later ) will be needed to execute txs.
```
Comment (weiyuan95) [inline README.md:147]:
```suggestion
Admin and Operational addresses will be Gnosis safes, ideally will be with at least 3 owners with a 2/3 quorum.
```

File: README.md [hunk 10]
```diff
@@ -127,3 +127,23 @@ BridgeV1 Contract address: [0xE029B5156c2e597c72f7c8D279411e1fD9a30126](https://
...
+
+More admins can be added later, for more info [Gnosis safe: adding owners](https://help.gnosis-safe.io/en/articles/3950657-add-owners).
```
Comment (weiyuan95) [inline README.md:149]:
```suggestion
More admins can be added later, for more info, refer to [Gnosis safe: adding owners](https://help.gnosis-safe.io/en/articles/3950657-add-owners).
```

File: packages/smartcontracts/src/hardhat.config.ts [hunk 11]
```diff
@@ -77,7 +77,7 @@ const config: HardhatUserConfig = {
   gasReporter: {
     currency: 'USD',
     // To enable gas report, set enabled to true
-    enabled: false,
+    enabled: true,
```
Comment (weiyuan95) [inline packages/smartcontracts/src/hardhat.config.ts:80]: unintended?

File: packages/smartcontracts/src/contracts/BridgeV1.sol [hunk 12]
```diff
@@ -427,7 +431,7 @@ contract BridgeV1 is UUPSUpgradeable, EIP712Upgradeable, AccessControlUpgradeabl
      * @param _amount Ideally will be the value of erc20 token
      * @return netAmountInWei net balance after the fee amount taken
      */
-    function amountAfterFees(uint256 _amount) private view returns (uint256 netAmountInWei) {
+    function amountAfterFees(uint256 _amount) internal view returns (uint256 netAmountInWei) {
```
Comment (weiyuan95) [inline packages/smartcontracts/src/contracts/BridgeV1.sol:434]: is this an intentional change?

File: README.md [hunk 13]
```diff
@@ -39,11 +39,11 @@ TODO
 
 To change the state of any smart contract, users need to approve the smart contract of the respective token via the `approve()` function first. Once approved, user will be able to bridge the token over to DefiChain.
 
-### Fund ERC20 tokens - to transfer ERC20 tokens from an EOA to the Bridge
+### Bridge ERC20 tokens - to transfer ERC20 tokens from an EOA to the Bridge
 
 Once approved, user will call the `bridgeToDeFiChain()` function with following arguments: `_defiAddress`- address on Defi Chain that receiving funds, `_tokenAddress` - ERC20 token's address and `_amount` amount to bridge over to Defi chain.
 
-### Fund Ether - to transfer ERC20 tokens from an EOA to the Bridge
+### Bridge Ether - to transfer ETHER from an EOA to the Bridge
```
Comment (weiyuan95) [inline README.md:46]: This section can be removed too after #276 🤣

**Review by weiyuan95 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (Abhishekkochar) [reply README.md:131]: I removed this on PR #289

File: packages/smartcontracts/src/contracts/BridgeV1.sol (hunk 12 above)
Comment (Abhishekkochar) [reply packages/smartcontracts/src/contracts/BridgeV1.sol:434]: Yes, In term of gas, both are the same. Only difference is visibility. When `private` only this contract can use this function. When `internal` inherited contracts can use that.

Comment (weiyuan95) [reply packages/smartcontracts/src/contracts/BridgeV1.sol:434]: Hmm, why would a contract need to inherit from BridgeV1? 🤔

**Review by weiyuan95 — APPROVED** [review]
This LGTM, just 1 quick question on the contract method visibility / inheritance

File: packages/smartcontracts/src/contracts/BridgeV1.sol (hunk 12 above)
Comment (Abhishekkochar) [reply packages/smartcontracts/src/contracts/BridgeV1.sol:434]: This function is only view, so no state change. Even if we put it to public, this won't change anything except the deployment cost.

**Pull request merged**